content/blogposts/2022/ranking-in-a-week.md

```diff
@@ -0,0 +1,120 @@
+
+At this point, the ask for PageRank deliverable becomes concrete, and we settle on an adequate, although incredibly brisk, timeline. On October 23rd, our effort begins in earnest.
+
+We begin by concentrating on the _construction_ of a graph over which we can run PageRank. Our chosen solution employs a background worker that [periodically exports code intelligence data](https://github.com/sourcegraph/sourcegraph/pull/43334) into CSV files in a GCS bucket. For each repository for which we have precise code intelligence data, we export several files containing data related to the tip of that repository's default branch.
+
+First, we output `documents.csv`, which correlates a repo and file path pair with a randomly generated identifier (increasing integers for the sake of simplicity in this example). These items correspond to the white vertices in the graph of text documents shown earlier in this post.
+
+```csv
+pathID,repo,path
+1,github.com/sourcegraph/blog-post-example,cmd/app/main.go
+2,github.com/sourcegraph/blog-post-example,cmd/audit/main.go
+3,github.com/sourcegraph/blog-post-example,internal/database.go
+4,github.com/sourcegraph/blog-post-example,internal/experiments.go
+5,github.com/sourcegraph/blog-post-example,internal/handler.go
+6,github.com/sourcegraph/blog-post-example,internal/init.go
+```
+
+Next, we output `references.csv`, which corresponds to the edges _between_ the white vertices. Such an edge is output when there is a direct reference from a variable defined in another file.
+
+```csv
+from,to
+1,3
+1,4
+2,5
+3,4
+4,3
+4,5
+5,3
+5,4
+6,3
+6,5
+```
+
+To support references between repositories, we output `monikers.csv`, which will declare a set of _monikers_ that are either declared in or referenced by a particular path. If one repository declares an `import` moniker, and another repository declares an `export` moniker with the same scheme and identifier, then we create an edge between the documents across repositories.
+
+```csv
+pathID,monikerType,schema,identifier
+1,import,gomod,github.com/sourcegraph/log/Logger
+1,import,gomod,github.com/sourcegraph/log/Logger.Log
+5,import,gomod,github.com/sourcegraph/log/Logger
+5,import,gomod,github.com/sourcegraph/log/Logger.Log
+6,import,gomod,github.com/sourcegraph/log/Logger
+6,import,gomod,github.com/sourcegraph/log/Logger.Log
+```
+
+We would later round out this implemetation by also [pruning irrelevant data](https://github.com/sourcegraph/sourcegraph/pull/43832) from the target GCS bucket as newer indexes for repositories replace older indexes. Since the landing of this export task, it has become the basis of several other solutions currently in-flight (details to come in future blog posts).
```

Comment:
```suggestion
We would later round out this implementation by also [pruning irrelevant data](https://github.com/sourcegraph/sourcegraph/pull/43832) from the target GCS bucket as newer indexes for repositories replace older indexes. Since the landing of this export task, it has become the basis of several other solutions currently in-flight (details to come in future blog posts).
```